taskvine/src/manager/vine_manager.c

```diff
@@ -1662,6 +1665,23 @@ static vine_result_code_t get_result(struct vine_manager *q, struct vine_worker_
 	/* Consume the stdout data immediately following this message. */
 	get_stdout(q, w, t, output_length);
 
+	/* If a library task fails, remove this instance  */
+	/* handle failed library instances */
+	if (task_status == VINE_RESULT_LIBRARY_FAILED) {
+		debug(D_VINE, "Task %d library %s failed", t->task_id, t->provides_library);
+		reset_task_to_state(q, t, VINE_TASK_RETRIEVED);
+		t->refcount--;
+		struct vine_task *original = hash_table_lookup(q->library_templates, t->provides_library);
```

Comment:
Careful here -- if there are enough failures, the original could be removed.  So you must check for `if(!original)` here and anywhere we do `hash_table_lookup(q->library_templates)`